engine/src/flutter/lib/gpu/lib/src/render_pass.dart

```diff
@@ -168,6 +168,27 @@ base class Scissor {
   }
 }
 
+base class DepthRange {
+  DepthRange({this.zNear = 0.0, this.zFar = 1.0});
+
+  double zNear;
+  double zFar;
+}
+
+base class Viewport {
+  Viewport({this.x = 0, this.y = 0, this.width = 0, this.height = 0, DepthRange? depthRange = null})
+    : this.depthRange = depthRange ?? DepthRange();
+
+  int x, y, width, height;
```

Comment:
Internally we take a floating rect, but using ints here is actually a more correct restriction here. So LGTM. 👍